src/server/pfs/server/driver.go

```diff
@@ -56,9 +57,13 @@ const (
 
 	// maxInt is the maximum value for 'int' (system-dependent). Not in 'math'!
 	maxInt = int(^uint(0) >> 1)
+)
 
-	// Maximum number of concurrent put object calls.
-	putObjectConcurrency = 100
+var (
+	// Memory limiter (Useful for limiting operations that could use a lot of memory)
+	memoryLimiter *semaphore.Weighted
+	// Limit the number of outstanding put object requests
+	putObjectLimiter = limit.New(100)
```

Comment:
Is there a reason to make these global variables rather than put them within the `driver`? I think we mostly only do one `driver` per process, except in tests, but having global state potentially shared between drivers seems like a bit of a funny smell.

Reply:
Well, it could be in the driver, but the idea behind this is that it is a global limiter in the container. I have a feeling we are going to want to change this over time as well because preventing OOM kills while also using the available memory is very tricky and this is not the most elegant solution.